packages/go/dawgs/util/errors.go

```diff
@@ -58,9 +60,14 @@ func (s *errorCollector) Combined() error {
 }
 
 func IsNeoTimeoutError(err error) bool {
-	if castError, ok := err.(*neo4j.Neo4jError); !ok {
+	switch e := err.(type) {
```

Comment:
Quick question here, does the `dawgs` package wrap the original Neo4j error or does it output its own? If it's wrapped we might be able to use the `errors` package to unwrap and get to the original error to tighten this logic up.

If not then this is best effort but it also may be worth seeing what the change looks like to get `dawgs` to wrap the originating error.

Reply:
Oh good call, I hadn't considered that. Let me play around with it locally and I'll clean it up if possible.

Reply:
I did some research here and I don't believe that wrapping would be an easy to implement solution here. The error is returned directly from Neo4j and put into a `graph.Error` which is a struct containing the query performed and the raw driver error. The underlying `neo4j-go-driver` library wraps some errors, but timeouts are not one of them.

I think I'll leave this as-is for now and we can look at a better system in a future PR.